packages/paste-core/utilities/grid/__tests__/grid.test.tsx

```diff
@@ -0,0 +1,429 @@
+
+describe('Grid Unit Tests', () => {
+  it('it should return: -0.25rem', (): void => {
+    const mockGutter: Space = 'space20';
+    expect(getOuterGutterPull(DefaultTheme, mockGutter)).toStrictEqual('-0.25rem');
+  });
+
+  it('it should return: -0.25rem, -0.5rem, -0.75rem', (): void => {
+    const mockGutter: Space = ['space20', 'space30', 'space40'];
+    expect(getOuterGutterPull(DefaultTheme, mockGutter)).toStrictEqual(['-0.25rem', '-0.5rem', '-0.75rem']);
+  });
+
+  it('it should return vertical: 0', (): void => {
+    const mockVertical = false;
+    expect(getStackedColumns(mockVertical)).toStrictEqual('0');
+  });
+
+  it('it should return vertical: 100%', (): void => {
+    const mockVertical = true;
+    expect(getStackedColumns(mockVertical)).toStrictEqual('100%');
+  });
+
+  it('it should return vertical: 100%, 0, 100%', (): void => {
+    const mockVertical = [true, false, true];
+    expect(getStackedColumns(mockVertical)).toStrictEqual(['100%', '0', '100%']);
+  });
+
+  it('it should return vertical: null, null, null', (): void => {
+    const mockVertical = ['test', 'test', 'test'];
+    expect(getStackedColumns(mockVertical)).toStrictEqual([null, null, null]);
+  });
+
+  it('it should return offset: 16.666666666666664%', (): void => {
+    const mockOffset = 2;
+    expect(getColumnOffset(mockOffset)).toStrictEqual('16.666666666666664%');
+  });
+
+  it('it should return offset: 16.666666666666664%, 25%, 50%', (): void => {
+    const mockOffset = [2, 3, 6];
+    expect(getColumnOffset(mockOffset)).toStrictEqual(['16.666666666666664%', '25%', '50%']);
+  });
+
+  it('it should return span: 50%', (): void => {
+    const mockSpan = {
+      count: 12,
+      span: 6,
+    };
+    expect(getColumnSpan(mockSpan)).toStrictEqual('50%');
+  });
+
+  it('it should return span: 25%', (): void => {
+    const mockSpan = {
+      count: 12,
+      span: 3,
+    };
+    expect(getColumnSpan(mockSpan)).toStrictEqual('25%');
+  });
+
+  it('it should return span: 33.33333333333333%', (): void => {
+    const mockSpan = {
+      count: 3,
+    };
+    expect(getColumnSpan(mockSpan)).toStrictEqual('33.33333333333333%');
+  });
+
+  it('it should return span: 8.333333333333332%', (): void => {
+    const mockSpan = {
+      count: 12,
+    };
+    expect(getColumnSpan(mockSpan)).toStrictEqual('8.333333333333332%');
+  });
+
+  it('it should return span: 8.333333333333332% (null span)', (): void => {
+    const mockSpan = {
+      span: null,
+    };
+    expect(getColumnSpan(mockSpan)).toStrictEqual('8.333333333333332%');
+  });
+
+  it('it should return span: 25%, 50%, 25%', (): void => {
+    const mockSpan = {
+      count: 12,
+      span: [3, 6, 3],
+    };
+    expect(getColumnSpan(mockSpan)).toStrictEqual(['25%', '50%', '25%']);
+  });
+});
+
+describe('Grid Gutter Prop', () => {
+  it('it should set a gutter property', (): void => {
```

Comment:
What does "property" mean, exactly? What are we expecting to see in the snapshot

Reply:
Test descriptions rewritten.